Add sanity checks to diagnostics

LisaBock opened the pull request:
This PR adds two diagnostics:
sanity_check_1: Timeseries of global sum of water vapour and anomly of global air mass
sanity_check_2: Timeseries of global means of several variables with red horizontal lines showing "reasonable" ranges for the global mean calculated from minimum and maximum global mean values found in reference datasets (observations, reanalyses) across all months and all reference datasets.

bettina-gier commented on hybridesmbench/eval/_diags/sanity_checks_mean/data/clivi/metadata.yml:
```diff
@@ -0,0 +1,40 @@
+? /home/b/b309059/HybridESMBench/hybridesmbench/eval/_diags/sanity_checks_mean/data/clivi/MultiModelMax_Amon_clivi_1982-2018.nc
```
```suggestion
? clivi/MultiModelMax_Amon_clivi_1982-2018.nc
```
Needs to be relative paths

bettina-gier commented on hybridesmbench/eval/_diags/sanity_checks_mean/data/clivi/metadata.yml:
```diff
@@ -0,0 +1,40 @@
+  dataset: MultiModelMax
+  diagnostic: clivi
+  end_year: 2018
+  filename: /home/b/b309059/HybridESMBench/hybridesmbench/eval/_diags/sanity_checks_mean/data/clivi/MultiModelMax_Amon_clivi_1982-2018.nc
```
```suggestion
  filename: clivi/MultiModelMax_Amon_clivi_1982-2018.nc
```

bettina-gier commented on hybridesmbench/eval/_diags/sanity_checks_mean/data/clivi/metadata.yml:
```diff
@@ -0,0 +1,40 @@
+  type: sat
+  units: kg m-2
+  variable_group: clivi
+? /home/b/b309059/HybridESMBench/hybridesmbench/eval/_diags/sanity_checks_mean/data/clivi/MultiModelMin_Amon_clivi_1982-2018.nc
```
```suggestion
? clivi/MultiModelMin_Amon_clivi_1982-2018.nc
```

bettina-gier commented on hybridesmbench/eval/_diags/sanity_checks_mean/data/clivi/metadata.yml:
```diff
@@ -0,0 +1,40 @@
+  dataset: MultiModelMin
+  diagnostic: clivi
+  end_year: 2018
+  filename: /home/b/b309059/HybridESMBench/hybridesmbench/eval/_diags/sanity_checks_mean/data/clivi/MultiModelMin_Amon_clivi_1982-2018.nc
```
```suggestion
  filename: clivi/MultiModelMin_Amon_clivi_1982-2018.nc
```

bettina-gier commented on hybridesmbench/eval/_diags/sanity_checks_mean/data/clt/metadata.yml:
```diff
@@ -0,0 +1,38 @@
+? /home/b/b309059/HybridESMBench/hybridesmbench/eval/_diags/sanity_checks_mean/data/clt/MultiModelMax_Amon_clt_1982-2018.nc
```
```suggestion
? clt/MultiModelMax_Amon_clt_1982-2018.nc
```

bettina-gier commented on hybridesmbench/eval/_diags/sanity_checks_mean/data/clt/metadata.yml:
```diff
@@ -0,0 +1,38 @@
+  dataset: MultiModelMax
+  diagnostic: clt
+  end_year: 2018
+  filename: /home/b/b309059/HybridESMBench/hybridesmbench/eval/_diags/sanity_checks_mean/data/clt/MultiModelMax_Amon_clt_1982-2018.nc
```
```suggestion
  filename: clt/MultiModelMax_Amon_clt_1982-2018.nc
```

bettina-gier commented on hybridesmbench/eval/_diags/sanity_checks_mean/data/clt/metadata.yml:
```diff
@@ -0,0 +1,38 @@
+  timerange: 1982/2018
+  units: '%'
+  variable_group: clt
+? /home/b/b309059/HybridESMBench/hybridesmbench/eval/_diags/sanity_checks_mean/data/clt/MultiModelMin_Amon_clt_1982-2018.nc
```
```suggestion
? clt/MultiModelMin_Amon_clt_1982-2018.nc
```

bettina-gier commented on hybridesmbench/eval/_diags/sanity_checks_mean/data/clt/metadata.yml:
```diff
@@ -0,0 +1,38 @@
+  dataset: MultiModelMin
+  diagnostic: clt
+  end_year: 2018
+  filename: /home/b/b309059/HybridESMBench/hybridesmbench/eval/_diags/sanity_checks_mean/data/clt/MultiModelMin_Amon_clt_1982-2018.nc
```
```suggestion
  filename: clt/MultiModelMin_Amon_clt_1982-2018.nc
```

bettina-gier commented on hybridesmbench/eval/_diags/sanity_checks_mean/data/hfls/metadata.yml:
```diff
@@ -0,0 +1,48 @@
+? /home/b/b309059/HybridESMBench/hybridesmbench/eval/_diags/sanity_checks_mean/data/hfls/OBS6_MERRA2_MultiModelMin_Amon_hfls_1980-2021.nc
```
```suggestion
? hfls/OBS6_MERRA2_MultiModelMin_Amon_hfls_1980-2021.nc
```

bettina-gier commented on hybridesmbench/eval/_diags/sanity_checks_mean/data/hfls/metadata.yml:
```diff
@@ -0,0 +1,48 @@
+  dataset: MERRA2
+  diagnostic: hfls
+  end_year: 2021
+  filename: /home/b/b309059/HybridESMBench/hybridesmbench/eval/_diags/sanity_checks_mean/data/hfls/OBS6_MERRA2_MultiModelMin_Amon_hfls_1980-2021.nc
```
```suggestion
  filename: hfls/OBS6_MERRA2_MultiModelMin_Amon_hfls_1980-2021.nc
```

bettina-gier commented on hybridesmbench/eval/_diags/sanity_checks_mean/data/hfls/metadata.yml:
```diff
@@ -0,0 +1,48 @@
+  units: W m-2
+  variable_group: hfls
+  version: 5.12.4
+? /home/b/b309059/HybridESMBench/hybridesmbench/eval/_diags/sanity_checks_mean/data/hfls/OBS6_MERRA2_MultiModelMax_Amon_hfls_1980-2021.nc
```
```suggestion
? hfls/OBS6_MERRA2_MultiModelMax_Amon_hfls_1980-2021.nc
```

bettina-gier commented on hybridesmbench/eval/_diags/sanity_checks_mean/data/hfls/metadata.yml:
```diff
@@ -0,0 +1,48 @@
+  dataset: MERRA2
+  diagnostic: hfls
+  end_year: 2021
+  filename: /home/b/b309059/HybridESMBench/hybridesmbench/eval/_diags/sanity_checks_mean/data/hfls/OBS6_MERRA2_MultiModelMax_Amon_hfls_1980-2021.nc
```
```suggestion
  filename: hfls/OBS6_MERRA2_MultiModelMax_Amon_hfls_1980-2021.nc
```

bettina-gier commented on hybridesmbench/eval/_diags/sanity_checks_mean/data/hfss/metadata.yml:
```diff
@@ -0,0 +1,48 @@
+? /home/b/b309059/HybridESMBench/hybridesmbench/eval/_diags/sanity_checks_mean/data/hfss/OBS6_MERRA2_MultiModelMin_Amon_hfss_1980-2021.nc
```
```suggestion
? hfss/OBS6_MERRA2_MultiModelMin_Amon_hfss_1980-2021.nc
```

bettina-gier commented on hybridesmbench/eval/_diags/sanity_checks_mean/data/hfss/metadata.yml:
```diff
@@ -0,0 +1,48 @@
+  dataset: MERRA2
+  diagnostic: hfss
+  end_year: 2021
+  filename: /home/b/b309059/HybridESMBench/hybridesmbench/eval/_diags/sanity_checks_mean/data/hfss/OBS6_MERRA2_MultiModelMin_Amon_hfss_1980-2021.nc
```
```suggestion
  filename: hfss/OBS6_MERRA2_MultiModelMin_Amon_hfss_1980-2021.nc
```

bettina-gier commented on hybridesmbench/eval/_diags/sanity_checks_mean/data/hfss/metadata.yml:
```diff
@@ -0,0 +1,48 @@
+  units: W m-2
+  variable_group: hfss
+  version: 5.12.4
+? /home/b/b309059/HybridESMBench/hybridesmbench/eval/_diags/sanity_checks_mean/data/hfss/OBS6_MERRA2_MultiModelMax_Amon_hfss_1980-2021.nc
```
```suggestion
? hfss/OBS6_MERRA2_MultiModelMax_Amon_hfss_1980-2021.nc
```

bettina-gier commented on hybridesmbench/eval/_diags/sanity_checks_mean/data/hfss/metadata.yml:
```diff
@@ -0,0 +1,48 @@
+  dataset: MERRA2
+  diagnostic: hfss
+  end_year: 2021
+  filename: /home/b/b309059/HybridESMBench/hybridesmbench/eval/_diags/sanity_checks_mean/data/hfss/OBS6_MERRA2_MultiModelMax_Amon_hfss_1980-2021.nc
```
```suggestion
  filename: hfss/OBS6_MERRA2_MultiModelMax_Amon_hfss_1980-2021.nc
```

bettina-gier commented on hybridesmbench/eval/_diags/sanity_checks_mean/data/pr/metadata.yml:
```diff
@@ -0,0 +1,38 @@
+? /home/b/b309059/HybridESMBench/hybridesmbench/eval/_diags/sanity_checks_mean/data/pr/MultiModelMin_Amon_pr_1979-2022.nc
```
```suggestion
? pr/MultiModelMin_Amon_pr_1979-2022.nc
```

bettina-gier commented on hybridesmbench/eval/_diags/sanity_checks_mean/data/pr/metadata.yml:
```diff
@@ -0,0 +1,38 @@
+  dataset: MultiModelMin
+  diagnostic: pr
+  end_year: 2022
+  filename: /home/b/b309059/HybridESMBench/hybridesmbench/eval/_diags/sanity_checks_mean/data/pr/MultiModelMin_Amon_pr_1979-2022.nc
```
```suggestion
  filename: pr/MultiModelMin_Amon_pr_1979-2022.nc
```

bettina-gier commented on hybridesmbench/eval/_diags/sanity_checks_mean/data/pr/metadata.yml:
```diff
@@ -0,0 +1,38 @@
+  timerange: 1979/2022
+  units: mm day-1
+  variable_group: pr
+? /home/b/b309059/HybridESMBench/hybridesmbench/eval/_diags/sanity_checks_mean/data/pr/MultiModelMax_Amon_pr_1979-2022.nc
```
```suggestion
? pr/MultiModelMax_Amon_pr_1979-2022.nc
```

bettina-gier commented on hybridesmbench/eval/_diags/sanity_checks_mean/data/pr/metadata.yml:
```diff
@@ -0,0 +1,38 @@
+  dataset: MultiModelMax
+  diagnostic: pr
+  end_year: 2022
+  filename: /home/b/b309059/HybridESMBench/hybridesmbench/eval/_diags/sanity_checks_mean/data/pr/MultiModelMax_Amon_pr_1979-2022.nc
```
```suggestion
  filename: pr/MultiModelMax_Amon_pr_1979-2022.nc
```

bettina-gier commented on hybridesmbench/eval/_diags/sanity_checks_mean/data/prw/metadata.yml:
```diff
@@ -0,0 +1,40 @@
+? /home/b/b309059/HybridESMBench/hybridesmbench/eval/_diags/sanity_checks_mean/data/prw/MultiModelMax_Amon_prw_1980-2021.nc
```
```suggestion
? prw/MultiModelMax_Amon_prw_1980-2021.nc
```

bettina-gier commented on hybridesmbench/eval/_diags/sanity_checks_mean/data/prw/metadata.yml:
```diff
@@ -0,0 +1,40 @@
+  dataset: MultiModelMax
+  diagnostic: prw
+  end_year: 2021
+  filename: /home/b/b309059/HybridESMBench/hybridesmbench/eval/_diags/sanity_checks_mean/data/prw/MultiModelMax_Amon_prw_1980-2021.nc
```
```suggestion
  filename: prw/MultiModelMax_Amon_prw_1980-2021.nc
```

bettina-gier commented on hybridesmbench/eval/_diags/sanity_checks_mean/data/prw/metadata.yml:
```diff
@@ -0,0 +1,40 @@
+  timerange: 1980/2021
+  units: kg m-2
+  variable_group: prw
+? /home/b/b309059/HybridESMBench/hybridesmbench/eval/_diags/sanity_checks_mean/data/prw/MultiModelMin_Amon_prw_1980-2021.nc
```
```suggestion
? prw/MultiModelMin_Amon_prw_1980-2021.nc
```

bettina-gier commented on hybridesmbench/eval/_diags/sanity_checks_mean/data/prw/metadata.yml:
```diff
@@ -0,0 +1,40 @@
+  dataset: MultiModelMin
+  diagnostic: prw
+  end_year: 2021
+  filename: /home/b/b309059/HybridESMBench/hybridesmbench/eval/_diags/sanity_checks_mean/data/prw/MultiModelMin_Amon_prw_1980-2021.nc
```
```suggestion
  filename: prw/MultiModelMin_Amon_prw_1980-2021.nc
```

bettina-gier commented on hybridesmbench/eval/_diags/sanity_checks_mean/data/rlut/metadata.yml:
```diff
@@ -0,0 +1,38 @@
+? /home/b/b309059/HybridESMBench/hybridesmbench/eval/_diags/sanity_checks_mean/data/rlut/MultiModelMin_Amon_rlut_1982-2022.nc
```
```suggestion
? rlut/MultiModelMin_Amon_rlut_1982-2022.nc
```

bettina-gier commented on hybridesmbench/eval/_diags/sanity_checks_mean/data/rlut/metadata.yml:
```diff
@@ -0,0 +1,38 @@
+  dataset: MultiModelMin
+  diagnostic: rlut
+  end_year: 2022
+  filename: /home/b/b309059/HybridESMBench/hybridesmbench/eval/_diags/sanity_checks_mean/data/rlut/MultiModelMin_Amon_rlut_1982-2022.nc
```
```suggestion
  filename: rlut/MultiModelMin_Amon_rlut_1982-2022.nc
```

bettina-gier commented on hybridesmbench/eval/_diags/sanity_checks_mean/data/rlut/metadata.yml:
```diff
@@ -0,0 +1,38 @@
+  timerange: 1982/2022
+  units: W m-2
+  variable_group: rlut
+? /home/b/b309059/HybridESMBench/hybridesmbench/eval/_diags/sanity_checks_mean/data/rlut/MultiModelMax_Amon_rlut_1982-2022.nc
```
```suggestion
? rlut/MultiModelMax_Amon_rlut_1982-2022.nc
```

bettina-gier commented on hybridesmbench/eval/_diags/sanity_checks_mean/data/rsut/metadata.yml:
```diff
@@ -0,0 +1,42 @@
+  dataset: MultiModelMin
+  diagnostic: rsut
+  end_year: 2022
+  filename: /home/b/b309059/HybridESMBench/hybridesmbench/eval/_diags/sanity_checks_mean/data/rsut/MultiModelMin_Amon_rsut_1982-2022.nc
```
```suggestion
  filename: rsut/MultiModelMin_Amon_rsut_1982-2022.nc
```

bettina-gier commented on hybridesmbench/eval/_diags/sanity_checks_mean/data/tas/metadata.yml:
```diff
@@ -0,0 +1,38 @@
+? /home/b/b309059/HybridESMBench/hybridesmbench/eval/_diags/sanity_checks_mean/data/tas/MultiModelMax_Amon_tas_1980-2021.nc
```
```suggestion
? tas/MultiModelMax_Amon_tas_1980-2021.nc
```

bettina-gier commented on hybridesmbench/eval/_diags/sanity_checks_mean/data/tas/metadata.yml:
```diff
@@ -0,0 +1,38 @@
+  dataset: MultiModelMax
+  diagnostic: tas
+  end_year: 2021
+  filename: /home/b/b309059/HybridESMBench/hybridesmbench/eval/_diags/sanity_checks_mean/data/tas/MultiModelMax_Amon_tas_1980-2021.nc
```
```suggestion
  filename: tas/MultiModelMax_Amon_tas_1980-2021.nc
```

bettina-gier commented on hybridesmbench/eval/_diags/sanity_checks_mean/data/tas/metadata.yml:
```diff
@@ -0,0 +1,38 @@
+  timerange: 1980/2021
+  units: K
+  variable_group: tas
+? /home/b/b309059/HybridESMBench/hybridesmbench/eval/_diags/sanity_checks_mean/data/tas/MultiModelMin_Amon_tas_1980-2021.nc
```
```suggestion
? tas/MultiModelMin_Amon_tas_1980-2021.nc
```

bettina-gier commented on hybridesmbench/eval/_diags/sanity_checks_mean/data/tas/metadata.yml:
```diff
@@ -0,0 +1,38 @@
+  dataset: MultiModelMin
+  diagnostic: tas
+  end_year: 2021
+  filename: /home/b/b309059/HybridESMBench/hybridesmbench/eval/_diags/sanity_checks_mean/data/tas/MultiModelMin_Amon_tas_1980-2021.nc
```
```suggestion
  filename: tas/MultiModelMin_Amon_tas_1980-2021.nc
```

bettina-gier commented on hybridesmbench/eval/_diags/sanity_checks_mean/data/tauu/metadata.yml:
```diff
@@ -0,0 +1,42 @@
+? /home/b/b309059/HybridESMBench/hybridesmbench/eval/_diags/sanity_checks_mean/data/tauu/OBS6_MultiModelMin_Amon_tauu_1871-2021.nc
```
```suggestion
? tauu/OBS6_MultiModelMin_Amon_tauu_1871-2021.nc
```

bettina-gier commented on hybridesmbench/eval/_diags/sanity_checks_mean/data/tauu/metadata.yml:
```diff
@@ -0,0 +1,42 @@
+  dataset: MultiModelMin
+  diagnostic: tauu
+  end_year: 2021
+  filename: /home/b/b309059/HybridESMBench/hybridesmbench/eval/_diags/sanity_checks_mean/data/tauu/OBS6_MultiModelMin_Amon_tauu_1871-2021.nc
```
```suggestion
  filename: tauu/OBS6_MultiModelMin_Amon_tauu_1871-2021.nc
```

bettina-gier commented on hybridesmbench/eval/_diags/sanity_checks_mean/data/tauu/metadata.yml:
```diff
@@ -0,0 +1,42 @@
+  type: reanaly
+  units: Pa
+  variable_group: tauu
+? /home/b/b309059/HybridESMBench/hybridesmbench/eval/_diags/sanity_checks_mean/data/tauu/OBS6_MultiModelMax_Amon_tauu_1871-2021.nc
```
```suggestion
? tauu/OBS6_MultiModelMax_Amon_tauu_1871-2021.nc
```

bettina-gier commented on hybridesmbench/eval/_diags/sanity_checks_mean/data/tauu/metadata.yml:
```diff
@@ -0,0 +1,42 @@
+  dataset: MultiModelMax
+  diagnostic: tauu
+  end_year: 2021
+  filename: /home/b/b309059/HybridESMBench/hybridesmbench/eval/_diags/sanity_checks_mean/data/tauu/OBS6_MultiModelMax_Amon_tauu_1871-2021.nc
```
```suggestion
  filename: tauu/OBS6_MultiModelMax_Amon_tauu_1871-2021.nc
```

bettina-gier commented on hybridesmbench/eval/_diags/sanity_checks_sum/__init__.py:
```diff
@@ -0,0 +1,266 @@
+)
+
+
+class SanityChecksDiagnostic(ESMValToolDiagnostic):
```
```suggestion
class SanityChecksDiagnostic(ESMValToolDiagnostic):
```
You're calling both sanity checks SanitChecksDiagnostic as a class, should this also be mean/sum?

bettina-gier commented on hybridesmbench/eval/_diags/sanity_checks_sum/data/prw/metadata.yml:
```diff
@@ -0,0 +1,40 @@
+? /home/b/b309059/HybridESMBench/hybridesmbench/eval/_diags/sanity_checks_sum/data/prw/MultiModelMax_Amon_prw_1980-2021.nc
```
```suggestion
? prw/MultiModelMax_Amon_prw_1980-2021.nc
```

bettina-gier commented on hybridesmbench/eval/_diags/sanity_checks_sum/data/prw/metadata.yml:
```diff
@@ -0,0 +1,40 @@
+  dataset: MultiModelMax
+  diagnostic: prw_global
+  end_year: 2021
+  filename: /home/b/b309059/HybridESMBench/hybridesmbench/eval/_diags/sanity_checks_sum/data/prw/MultiModelMax_Amon_prw_1980-2021.nc
```
```suggestion
  filename: prw/MultiModelMax_Amon_prw_1980-2021.nc
```

bettina-gier commented on hybridesmbench/eval/_diags/sanity_checks_sum/data/prw/metadata.yml:
```diff
@@ -0,0 +1,40 @@
+  timerange: 1980/2021
+  units: kg
+  variable_group: prw
+? /home/b/b309059/HybridESMBench/hybridesmbench/eval/_diags/sanity_checks_sum/data/prw/MultiModelMin_Amon_prw_1980-2021.nc
```
```suggestion
? prw/MultiModelMin_Amon_prw_1980-2021.nc
```

bettina-gier commented on hybridesmbench/eval/_diags/sanity_checks_sum/data/prw/metadata.yml:
```diff
@@ -0,0 +1,40 @@
+  dataset: MultiModelMin
+  diagnostic: prw_global
+  end_year: 2021
+  filename: /home/b/b309059/HybridESMBench/hybridesmbench/eval/_diags/sanity_checks_sum/data/prw/MultiModelMin_Amon_prw_1980-2021.nc
```
```suggestion
  filename: prw/MultiModelMin_Amon_prw_1980-2021.nc
```

LisaBock commented:
@bettina-gier , thanks for the review!
I changed the filenames in the metadata.yml files and also renamed the classes as you suggested.
For now, I would like to keep the "MultiModel" as these files are preprocessed by ESMValTool and named by the  "multi_model_statistics" preprocessor.
